yt_dlp/extractor/xanimu.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+from ..utils import int_or_none
+from .common import InfoExtractor
+
+
+class XanimuIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?xanimu\.com/(?:(?P<id>\d+)-)?(?P<display_id>[^/]+)/?'
+    _TESTS = [{
+        'url': 'https://xanimu.com/51944-the-princess-the-frog-hentai/',
+        'md5': '899b88091d753d92dad4cb63bbf357a7',
+        'info_dict': {
+            'id': '51944',
+            'display_id': 'the-princess-the-frog-hentai',
+            'ext': 'mp4',
+            'title': 'The Princess + The Frog Hentai',
+            'thumbnail': 'https://xanimu.com/storage/2020/09/the-princess-and-the-frog-hentai.jpg',
+            'description': r're:^Enjoy The Princess \+ The Frog Hentai',
+            'duration': 207.0,
+            'age_limit': 18
+        }
+    }, {
+        'url': 'https://xanimu.com/huge-expansion/',
+        'only_matching': True
+    }]
```

Comment:
does this not have a numeric id?

Reply:
No, it appears some videos like this one for example, don't have a numeric id in the url. This is why I made that part of the _VALID_URL regex optional.

Reply:
If not all videos have it, I suspect it may also not be unique. In that case, it would be better to always use the whole slug as id